configUtils.go

```diff
@@ -246,27 +246,101 @@ func (a *ConfigAnalyzer) CallChanges(current interface{}, future interface{}) (i
 				}
 				continue
 			}
-			if !futval.IsValid() {
-				fmt.Printf("futval is not valid!\n")
-			}
-			if futval.Interface() != fieldval.Interface() {
-				identical = false
-				noaction = false
-				// different values - add to list
-				group, ok := field.Tag.Lookup(a.configTag)
-				if ok {
-					c, ok2 := allchanges[group]
-					if !ok2 {
-						c = new(changes)
-						allchanges[group] = c
-						c.configgroup = group
+			// For slices we treat the slice as a single value,
+			// unless the slice is a slice of struct or ptr to struct
```

Comment:
It bothers me that we can now have an object that is _either_ a slice, or contain a set of slices.

Would it be possible to instead always return a set of slices, even if that set only has one member?

Reply:
Imo, we should avoid actively determining the structure of an object at runtime, as I feel that could lead to issues down the line.

Reply:
Talked w/ @SenRamakri, and he explained this.